awx_collection/plugins/module_utils/tower_module.py

```diff
@@ -112,6 +112,23 @@ def __init__(self, argument_spec=None, direct_params=None, error_callback=None,
         except Exception as e:
             self.fail_json(msg="Unable to resolve tower_host ({1}): {0}".format(hostname, e))
 
+    def build_url(self, endpoint, query_params=None):
+        # Make sure we start with /api/vX
+        if not endpoint.startswith("/"):
+            endpoint = "/{0}".format(endpoint)
+        if not endpoint.startswith("/api/"):
+            endpoint = "/api/v2{0}".format(endpoint)
+        if not endpoint.endswith('/') and '?' not in endpoint:
+            endpoint = "{0}/".format(endpoint)
+
+        # Update the URL path with the endpoint
+        url = self.url._replace(path=endpoint)
+
+        if query_params:
+            url = url._replace(query=urlencode(query_params))
+
+        return url
```

Comment:
Here again, trying to make `TowerModule` class the owner of any gymnastics that has to go on for `self.url`